index.js

```diff
@@ -3,19 +3,32 @@
 const url = require('url')
 const fp = require('fastify-plugin')
 const zipkin = require('zipkin')
+const assert = require('assert')
+const { Tracer, BatchRecorder, jsonEncoder: { JSON_V2 } } = require('zipkin')
+const { HttpLogger } = require('zipkin-transport-http')
+const CLSContext = require('zipkin-context-cls')
 
 const Some = zipkin.option.Some
 const None = zipkin.option.None
 const Instrumentation = zipkin.Instrumentation
 
 function zipkinPlugin (fastify, opts, next) {
-  const tracer = opts.tracer
-  if (tracer == null) return next(new Error('No tracer specified'))
+  assert(opts.serviceName, 'serviceName option should not be empty')
+
+  const recorder = new BatchRecorder({
+    logger: new HttpLogger({
+      endpoint: `${opts.zipkinUrl}/api/v2/spans`,
```

Comment:
is not mandatory this parameter?

Reply:
It is, so, added a test for it, and an assertion in case a value is not set.